### ISSUE TITLE
Add a way to whitelist non-snake case method names 

### ISSUE THREAD

**Issue by mundschenk-at** [WordPress-Coding-Standards/WordPress-Coding-Standards#851]
When applying the WPCS to PHPUnit tests, the `setUp` and `tearDown` methods are recognized as not conforming to snake case rules. There should be a way to whitelist those methods (as far as I could see whitelisting is currently only possible for variable names).

**Comment by JDGrimes** [comment]
Yeah, in my custom ruleset, I have this:

```xml
	<rule ref="WordPress.NamingConventions.ValidFunctionName">
		<exclude-pattern>/tests/*</exclude-pattern><!-- because of PHPUnit method names -->
	</rule>
```

Actually, I thought that this was part of WPCS's ruleset, but I guess it's not.

**Comment by jrfnl** [comment]
@mundschenk-at Could you post a minimal example of code which triggers that error in the case of a PHPUnit implementation ?

The sniff checks whether a class extends another class and/or implements an interface and bows out before the snakecase check if that's the case, so if your tests are based on PHPUnit you shouldn't have a problem as the test classes generally are declared as `class My_Test extends PHPUnit_Framework_TestCase {}`

**Comment by mundschenk-at** [comment]
@jrfnl Huh, it looks like something strange is going on with Atom's `linter-phpcs` package. It generally applies the same rules (via custom `phpcs.xml` but apparently triggers some false alarms? I can't reproduce the error on the command line at all.

**Comment by jrfnl** [comment]
@mundschenk-at In that case I suggest you check with the creators of Atom's `linter-phpcs` whether it includes the latest PHPCS version as an outdated PHPCS version is the only thing which comes to my mind as a possible reason for this inconsistency.

**Comment by mundschenk-at** [comment]
@jrfnl The weird thing is that it is using the command line tool (which is at version 2.8.0).

**Comment by jrfnl** [comment]
@mundschenk-at That is weird indeed. All the same, as you confirmed that the error does not appear when running WPCS from the command line, it still sounds like the problem is with `linter-phpcs`, not with WPCS.

**Comment by mundschenk-at** [comment]
Yes. I'll close the issue.

**Comment by jrfnl** [comment]
Hope you'll find a solution. If it turns out there is something we can do to assist from within WPCS, please feel free to re-open.
